content/client-lib-development-guide/features.textile

```diff
@@ -606,6 +612,13 @@ h4. ClientOptions
 *** @(TO3k3)@ @realtimeHost@ string - for development environments only; allows a non-default Ably Realtime host to be specified
 *** @(TO3k4)@ @port@ integer - for development environments only; allows a non-default Ably non-TLS port to be specified
 *** @(TO3k5)@ @tlsPort@ integer - for development environments only; allows a non-default Ably TLS port to be specified
+** @(TO3l)@ The follow attributes, if set, are used to change the default behaviour of the libary:
+*** @(TO3l1)@ @disconnectedRetryFrequency@ integer - default 15,000 (15s). When the connection is in the @DISCONNECTED@ state, this delay in milliseconds controls how frequently the client library attempts to reconnect automatically
+*** @(TO3l2)@ @suspendedRetryFrequency@ integer - default 30,000 (30s). When the connection is in the @SUSPENDED@ state, this delay in milliseconds controls how frequently the client library attempts to reconnect automatically
+*** @(TO3l3)@ @httpOpenTimeout@ integer - default 4,000 (4s). Timeout for opening the connection, available in the client library if supported by the transport
+*** @(TO3l4)@ @httpRequestTimeout@ integer - default 15,000 (15s). Timeout for any single HTTP request and response
+*** @(TO3l5)@ @httpMaxRetryCount@ integer - default 3. Max number of fallback host retries for HTTP requests that fail due to network issues or server problems
+*** @(TO3l6)@ @httpMaxRetryDuration@ integer - default 10,000 (10s). Max elapsed time in which fallback host retries for HTTP requests will be attempted i.e. if the first default host attempt takes 5s, and then the subsequent fallback retry attempt takes 7s, no further fallback host attempts will be made as the total elapsed time of 12s exceeds the default 10s limit
```

Comment:
Right, sorry, I'd misunderstood.

I'm unsure how to implement that in ably-js though. I might be missing something, but I can't see a way for a browser to access to the underlying TCP connection of an xhr request, only the http readyState. That doesn't have anything in between `1` (`OPENED`, ie you've called `open`, but doesn't imply anything's succeeded, you get `1` even if you call `open` with a nonexistant domain) and `2` (`HEADERS_RECEIVED`, ie response's http status is available -- by which time it'll have started sending a response, which is what cancels the `httpRequestTimeout`, presumably it isn't a common failure mode to send headers and then no response). https://xhr.spec.whatwg.org/#dom-xmlhttprequest-readystate